### PR TITLE
Add security events for access token deletions

### PR DESCRIPTION
**Pull request by willdollman**
## Test plan

- [x] New unit tests
- [x] Manual testing locally - tested `/.api/graphql?DeleteAccessToken` with `byID` and `byToken`

### CODE REVIEW

File: internal/database/access_tokens_test.go [hunk 1]
```diff
@@ -89,10 +90,74 @@ func TestAccessTokens_Create(t *testing.T) {
...
+		Password:              "p2",
+		EmailVerificationCode: "c2",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create context with valid actor; required by logging
+	subjectActor := actor.FromUser(subject.ID)
+	ctxWithActor := actor.WithActor(context.Background(), subjectActor)
```
Comment (willdollman) [inline internal/database/access_tokens_test.go:122]: Unlike the `Create` methods, the delete methods have to retrieve the UID from the context - is this the recommended way to set that for tests?

File: internal/database/access_tokens.go [hunk 2]
```diff
@@ -406,6 +447,19 @@ func toSHA256Bytes(input []byte) []byte {
 	return b[:]
 }
 
+func (s *accessTokenStore) logAccessTokenDeleted(ctx context.Context, deletionType SecurityEventName, arg []byte) {
+	a := actor.FromContext(ctx)
+
+	securityEventStore := NewDBWith(s.logger, s).SecurityEventLogs()
+	securityEventStore.LogEvent(ctx, &SecurityEvent{
+		Name:      deletionType,
+		UserID:    uint32(a.UID),
```
Comment (willdollman) [inline internal/database/access_tokens.go:456]: I was caught out by `LogEvent` silently failing if `a.UID` was a zero value while writing unit tests. Do you think it's worth adding a warning log line in `LogEvent` when this happens?

Comment (vrto) [reply internal/database/access_tokens.go:456]: Yeah, I'd do that. In theory, we should always have the actor ID available, but if there was some kind of background job pruning stale tokens using internal actor the call would crash.

File: internal/database/access_tokens_test.go (hunk 1 above)
Comment (vrto) [reply internal/database/access_tokens_test.go:122]: Yep, looking good!

**Review by vrto — APPROVED** [review]
Good job tackling this, all good!

File: internal/database/access_tokens.go (hunk 2 above)
Comment (willdollman) [reply internal/database/access_tokens.go:456]: Realised what's going on - it _does_ log if the insert fails, but that type of logging isn't printed to screen during unit test runs (at least in the context I was running the tests).

https://github.com/sourcegraph/sourcegraph/blob/will/auditlog/security-events/access-token-deleted/internal/database/security_event_logs.go#L144

**Pull request merged**